src/test/java/teammates/test/cases/webapi/GetSessionResultsAsCsvActionTest.java

```diff
@@ -0,0 +1,377 @@
+package teammates.test.cases.webapi;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.NullHttpParameterException;
+import teammates.common.exception.RequestExceedingRangeException;
+import teammates.common.util.Const;
+import teammates.test.driver.CsvChecker;
+import teammates.ui.webapi.action.CsvResult;
+import teammates.ui.webapi.action.GetSessionResultsAsCsvAction;
+
+/**
+ * SUT: {@link GetSessionResultsAsCsvAction}.
+ */
+public class GetSessionResultsAsCsvActionTest extends BaseActionTest<GetSessionResultsAsCsvAction> {
```

Comment:
For this, what about a test for:
- typical case with section name "All"? (the if case above)
- typical case with stats shown?

Reply:
I have added the test cases for stats hidden and shown. Thanks for spotting that! Regarding the section name "All" I think it would better to remove the hardcoded value.

Reply:
I discovered a problem when I tried to add test for case with stats shown. I notice that all questions in the typical data bundle are text-based questions which have no statistics to show. I am considering adding a new session to the typical data bundle, but that will break other tests that also rely on the typical data bundle. May I have some suggestions on how to resolve this issue?

Reply:
@xpdavid is there any way to update the typical bundle? Or is it better to just create something just for this test here?

Reply:
If you don't want to create a separate json file (which in most of case is not necessary), you can manipulate/add/delete the database in `@BeforeMethod`

Reply:
I tried to replace the `typicalDataBundle` with a custom json file by doing the following: 
```Java
@BeforeMethod
public void setup() {
    dataBundle = loadDataBundle("/FeedbackSessionResultsCsvActionTest.json");
    removeAndRestoreDataBundle(dataBundle);
}
```
The json file has a newly added session named `Session with MCQ`. The sessions attributes can be accessed by `dataBundle.feedbackSessions.get("sessionWithMcq")`. Yet, the following exception where still thrown during the test:
```
Trying to view a non-existent feedback session: idOfTypicalCourse1/Session with MCQ
```
Did I miss anything?

Reply:
Interesting...